src/request-sender.spec.ts

```diff
@@ -34,12 +34,30 @@ describe('RequestSender', () => {
                 credentials: true,
                 headers: {
                     Accept: 'application/json, text/plain, */*',
-                    'Content-Type': 'application/json',
                 },
                 method: 'GET',
             });
         });
 
+        it('creates a HTTP request with content-type if payload is provided', () => {
```

Comment:
Do you need a negative test for this?

Reply:
At the moment, the negative case is covered by other cases. But it's implicit. I can make a new test to cover it explicitly.

Reply:
@PascalZajac I've added e5a2898 to cover the negative case.